charts/istio-alerts/templates/service-prometheusrule.yaml

```diff
@@ -1,57 +1,71 @@
-{{ $values           := .Values }}
-{{ $release          := .Release }}
-{{ if .Values.serviceRules.enabled }}
-
+{{- $values     := .Values }}
+{{- $release    := .Release }}
+{{- if .Values.serviceRules.enabled }}
 apiVersion: monitoring.coreos.com/v1
 kind: PrometheusRule
 metadata:
-  name: {{ .Release.Name }}-service-rules
+  name: {{ .Release.Name }}-{{ lower .Chart.Name }}-service-rules
   annotations:
     nextdoor.com/chart: {{ .Values.chart_name }}
     nextdoor.com/source: {{ .Values.chart_source }}
 spec:
   groups:
-  - name: {{ .Release.Name }}.{{ .Release.Namespace }}.serviceRules
+  - name: {{ .Release.Namespace }}.{{ .Release.Name }}.{{ .Chart.Name }}.IstioServiceRules
     rules:
-    {{ with .Values.serviceRules.http5XXMonitor }}
-    {{ if .enabled }}
-    - alert: {{ $release.Name }}-5xx-Rate-Too-High
+    {{- with .Values.serviceRules.http5XXMonitor }}
+    {{- if .enabled }}
+    - alert: 5xx-Rate-Too-High
       annotations:
-        summary: {{`High rate of 5xx errors from {{$labels.destination_service_name}} in namespace {{$labels.namespace}}`}}
+        summary: >-
+          {{`{{$labels.destination_service_name}} is throwing too many 5xx errors (namespace: {{$labels.namespace}})`}}
         runbook_url: {{ $values.defaults.runbookUrl }}#5xx-Rate-Too-High
         description: >-
-          High rate of 5xx responses from the {{`$labels.destination_service_name`}} VirtualService in namespace {{`{{$labels.namespace}}`}}.
+          High rate of 5xx responses from the {{`{{$labels.destination_service_name}}`}} VirtualService
```

Comment:
I reformatted this line and discovered bugs in the templating.

```
{{`$labels.destination_service_name`}}
```

needed to be
```
{{`{{$labels.destination_service_name}}`}}
```

The
```
{{`
```

tells Helm to stop processing this portion for templating... and we use that so that the final templated output is _a new template_ that will be processed by the alertmanager:

```
    - alert: 5xx-Rate-Too-High
      annotations:
        summary: >-
          {{$labels.destination_service_name}} is throwing too many 5xx errors (namespace: {{$labels.namespace}})
        runbook_url: https://github.com/Nextdoor/k8s-charts/blob/main/charts/istio-alerts/runbook.md#5xx-Rate-Too-High
        description: >-
          High rate of 5xx responses from the {{$labels.destination_service_name}} VirtualService
          in namespace {{$labels.namespace}}.
```
```

Reply:
yo dawg I heard you like templates...

Reply:
😮‍💨